### ISSUE TITLE
Allow defining search indexes as component attribute

### ISSUE THREAD

**Issue by josephguillaume** [pod-os/PodOS#36]
Currently, pos-navigation-bar only shows search when logged in, and only builds a global search index using indexes specified in a user's profile.

https://github.com/pod-os/PodOS/blob/dec48f25ed1b9cb07698b29ad2eeecb9dbfc1c08/elements/src/components/pos-navigation-bar/pos-navigation-bar.tsx#L31-L32

As a developer-user, in an app I might want to limit search to a specific set of resources, e.g. feed items/video titles, or individual photo captions. I don't necessarily want these in a global search index in other apps - given they can be many, they can "pollute" the search results.

I suggest the following interface:
```
<pos-navigation-bar search-indexes="./path/one/index.ttl ./path/two/index.ttl">
```
An alternative would be
```
<pos-navigation-bar search-profile="./path/to/custom/profile#this">
```
Where profile#this would have a label index predicate (though privateLabelIndex is probably not the right name). This would allow a larger set of search indices, but involves an additional layer of indirection.

This interface could also be public. This would allow search in a public wiki or issue tracker. The developer-user is responsible for ensuring the index files are available to public or only when logged in.

**Comment by angelo-v** [comment]
I agree the search component could be generalized. Currently `pos-navigation-bar` is a very powerful global search and navigation component similar to a browsers address bar. This should not change, but a more generic `pos-seach-bar` with an interface similar to what you describe could be extracted and reused within `pos-navigation-bar`.
